src/core/email_service.py

```
@@ -0,0 +1,416 @@
+from ok.util.file import get_relative_path, read_json_file, write_json_file
+
+from src.core.email_config import DEFAULT_EMAIL_CONFIG, EMAIL_CONFIG_NAME
+
+# 项目根目录（src/core/email_service.py -> 上三级为项目根）
+ROOT = Path(__file__).resolve().parents[2]
+HTML_DIR = ROOT / "html"
+DEFAULT_TEMPLATE_NAME = "test_email.html"
+
+
+def _safe_int(value, default: int) -> int:
+    """把配置值安全转成 int；非数字/None/空串时回退默认值，避免阻塞或报错。"""
```

Ruff (0.16.1)

[warning] Docstring contains ambiguous `；` (FULLWIDTH SEMICOLON). Did you mean `;` (SEMICOLON)?

(RUF002)